drivers/src/mailbox.rs

```diff
@@ -390,19 +406,68 @@ impl MailboxRecvTxn<'_> {
     ///
     pub fn recv_request(&mut self, data: &mut [u32]) -> CaliptraResult<()> {
         self.copy_request(data)?;
-        self.complete(true)?;
-        Ok(())
+        self.complete(true)
     }
 
+    /// Sends `data.len()` bytes to the mailbox FIFO
+    /// Transitions from Execute --> RdyForData
     ///
-    /// Transitions from Execute --> Idle
+    /// # Arguments
     ///
-    pub fn complete(&mut self, success: bool) -> CaliptraResult<()> {
+    /// * `data` - data buffer.
+    ///
+    /// # Returns
+    ///
+    /// Status of Operation
+    ///
+    fn copy_response(&mut self, data: &[u8]) -> CaliptraResult<()> {
         if self.state != MailboxOpState::Execute {
             return Err(CaliptraError::DRIVER_MAILBOX_INVALID_STATE);
         }
+
+        if !data.is_empty() {
```

Comment:
Is this actually necessary? To the hardware, sending a 0-byte response is different than sending a regular completed response.

Reply:
Oh sure if we'd rather that be the case I can absolutely drop this check. I wasn't sure if "DataReady" with zero size was a case we considered valid or if it was better to prevent that from happening.

Reply:
I think we should be able to remove this, and then add a separate test-case for the 0-byte response vs the completed (no data) response.